src/code/z_map_disp.c

```diff
@@ -1,109 +1,1877 @@
+
+                sPauseDungeonMap.roomSprite[sceneRoomIter] = sPauseDungeonMap.mapI_roomTextures[dungeonMapRoomIter];
+            }
+        }
+    }
+    return heapNext;
+}
+
+s32 MapDisp_IsOnStorey(s32 storey, f32 checkY) {
+    if (storey == 0) {
+        if ((sMapDisp.storeyYList[0] <= checkY) && ((sMapDisp.numStoreys == 1) || (checkY < sMapDisp.storeyYList[1]))) {
+            return true;
+        }
+    } else if (storey >= (sMapDisp.numStoreys - 1)) {
+        if (sMapDisp.storeyYList[sMapDisp.numStoreys - 1] <= checkY) {
+            return true;
+        }
+    } else if ((sMapDisp.storeyYList[storey] <= checkY) && (checkY < sMapDisp.storeyYList[storey + 1])) {
+        return true;
+    }
+    return false;
+}
+
+s32 MapDisp_ConvertBossSceneToDungeonScene(s32 sceneId) {
+    switch (sceneId) {
+        case SCENE_MITURIN_BS:
+            return SCENE_MITURIN;
+
+        case SCENE_HAKUGIN_BS:
+            return SCENE_HAKUGIN;
+
+        case SCENE_SEA_BS:
+            return SCENE_SEA;
+
+        case SCENE_INISIE_BS:
+            return SCENE_INISIE_N;
+
+        default:
+            return sceneId;
+    }
+}
+
+/**
+ * @brief Draws the dungeon room sprites for the pause menu dungeon map
+ *
+ * @param play
+ * @param viewX top left x position of the dungeon map view window
+ * @param viewY top left y posiiton of the dungeon map view window
+ * @param viewWidth width in pixels of the dungeon map view window
+ * @param viewHeight height in pixels of the dungeon map view window
+ * @param scaleFrac ratio to convert world space coordinates to map coordinates
+ * @param dungeonIndex enum DungeonIndex for retrieving map/compass data
+ */
+void MapDisp_DrawRooms(PlayState* play, s32 viewX, s32 viewY, s32 viewWidth, s32 viewHeight, f32 scaleFrac,
+                       s32 dungeonIndex) {
+    static u16 sUnvisitedRoomPal[0x10] = {
+        0x0000, 0x0000, 0xFFC1, 0x07C1, 0x07FF, 0x003F, 0xFB3F, 0xF305,
+        0x0453, 0x0577, 0x0095, 0x82E5, 0xFD27, 0x7A49, 0x94A5, 0x0001,
+    }; // palette 0
+    static u16 sVisitedRoomPal[0x10] = {
+        0x0000, 0x027F, 0xFFC1, 0x07C1, 0x07FF, 0x003F, 0xFB3F, 0xF305,
+        0x0453, 0x0577, 0x0095, 0x82E5, 0xFD27, 0x7A49, 0x94A5, 0x0001,
+    }; // palette 1
+    static u16 sCurrentRoomPal[0x10] = {
+        0x0000, 0x0623, 0xFFC1, 0x07C1, 0x07FF, 0x003F, 0xFB3F, 0xF305,
+        0x0453, 0x0577, 0x0095, 0x82E5, 0xFD27, 0x7A49, 0x94A5, 0x0001,
+    }; // palette 2
+    PauseContext* pauseCtx = &play->pauseCtx;
+    s32 pad[4];
+    s32 i;
+    s32 green = ((sPauseDungeonMap.animTimer * -120.0f / 40.0f) + 200.0f) * 31.0f / 255.0f;
+    s32 blue = ((sPauseDungeonMap.animTimer * 115.0f / 40.0f) + 140.0f) * 31.0f / 255.0f;
+
+    sCurrentRoomPal[1] = (green << 6) | (blue << 1) | 1;
+
+    if (CHECK_DUNGEON_ITEM(DUNGEON_MAP, dungeonIndex)) {
+        s32 requiredScopeTemp;
+
+        sUnvisitedRoomPal[0xF] = 0xAD5F;
```

Comment:
```suggestion
        sUnvisitedRoomPal[15] = 0xAD5F;
```
Small nit but these seem better as decimal.